### ISSUE TITLE
Search is broken on macOS/Safari 15

### ISSUE THREAD

**Issue by petecooper** [textpattern/textpattern.github.io#192]
STR: search docs site for anything, no results are shown. For example:

https://docs.textpattern.com/search/?q=form

macOS/Safari 15.0 shows the following console log:

```
[Error] QuotaExceededError: The quota has been exceeded.
	setItem (search.js:1:119221)
	(anonymous function) (search.js:1:119221)
	i (search.js:1:28112)
	fireWith (search.js:1:28876)
	l (search.js:1:79576)
	(anonymous function) (search.js:1:82019)
```

Likely fixable with https://github.com/textpattern/textpattern.github.io/issues/162 but if we can bodge something in the meantime, that would be super.

**Comment by philwareham** [comment]
No issues at upstream project, but to be honest that project has been abandoned for a number of years.

Worth consideration: https://github.com/christian-fei/Simple-Jekyll-Search

I don't have time to implement until at least w/c 1st November. If someone else wants to try a fix, then by all means have at it.

Also, would be nice to get the self-hosted site up in the near future and end our reliance on GitHub Pages.

**Comment by jools-r** [comment]
I wanted to mention this too. Likewise `QuotaExceededError: The quota has been exceeded` in Safari 15 and still working on Brave (i.e. Chrome).

**Comment by philwareham** [comment]
Will try to fix on Monday, can spare an hour that day.

**Comment by jools-r** [comment]
May I nudge on this one too?

**Comment by philwareham** [comment]
Yeah, I can't fix this without the Jekyll site being self-hosted. Basically, Lunr.js tries to build the search using `LocalStorage` but because the site is quite large it's exceeding Safari's default local storage limit.

If the site were self-noted, we could use a proper Ruby plugin to achieve search.

**Comment by petecooper** [comment]
I have issues with getting it to rebuild non-interactively, I am working on it.

**Comment by petecooper** [comment]
Docs site is now self-hosted.

**Comment by jools-r** [comment]
Could I nudge this one once more? It looks like you've added the packages and that the site is now self-hosted. I guess a puzzle piece is missing?

**Comment by philwareham** [comment]
@petecooper Hi Pete, I assume we use the `self-hosted` branch for development now (for the self hosted docs on our own server)?

**Comment by philwareham** [comment]
For investigation: https://www.algolia.com/developers/code-exchange/integrate-jekyll-with-algolia/

**Comment by philwareham** [comment]
TBH Jekyll seems fair dead nowadays. I guess we could look at Eleventy or just bring the docs into a Textpattern installation itself - don't know how we could maintain user editable pages on GitHub along with Textpattern database though?

**Comment by petecooper** [comment]
There's an issue somewhere (don't ask where) that I have to track viability of docs hosting on a docs-centric, non-GitHub service…I need to find it, and I'll link it, and we'll make it happen…it's been a while, so I'll go searching.

Getting Textpattern docs into Textpattern in a non-kludgy, maintainable way is a whole heap of work, and I'm inclined to use a reputable service with good search, being mindful of the huge amount of work @wion & @Bloke put into docs (and not undoing that).

Textpattern isn't the right platform for our docs, at least not in the sense that it can plug into GitHub and maintain syncing, we're not there yet. There's no shame in using other tools, we're progressive enough to avoid NIH syndrome, and if it lets us divert energies into core then that's a heavy tick in the box from me.

**Comment by petecooper** [comment]
Tagging https://github.com/textpattern/textpattern.github.io/issues/212
